java/core/src/java/org/apache/orc/impl/RedBlackTree.java

```diff
@@ -271,10 +271,14 @@ private boolean add(int node, boolean fromLeft, int parent,
 
   /**
    * Add the new key to the tree.
+   *
+   * @param bytes
+   * @param offset
+   * @param length
    * @return true if the element is a new one.
    */
-  protected boolean add() {
-    add(root, false, NULL, NULL, NULL);
+  protected boolean doAdd(byte[] bytes, int offset, int length) {
```

Comment:
Is this renamed to avoid conflicts?

Reply:
Yes.  There is already a `public` method with the same name/signature.